### DISCUSSION TITLE
Feature: 请求添加更多对社区插件、适配器的可用程度测试

### DISCUSSION THREAD

**Discussion by coralZone** [nonebot/discussions#479] [问答] [answered]
### 希望能解决的问题

- 希望能够对于官方推荐插件或是官方维护插件集成进CI,能更好查看插件的稳定性,适配最大版本

### 描述所需要的功能

- 构建完善的生态系统

**Comment by he0119** [comment]
请问有没有比较具体的例子呢？

**Comment by HuanLinOTO** [comment]
对于这样或者那样的问题，我建议提出这样的问题，然后提出那样的做法，以提高这样那样的效率，避免这样那样的问题重复发生

~~以及这个项目叫 `N O N E B O T` 不是 `N O N B O T`(~~

**Comment by coralZone** [comment]
> 请问有没有比较具体的例子呢？



> 对于这样或者那样的问题，我建议提出这样的问题，然后提出那样的做法，以提高这样那样的效率，避免这样那样的问题重复发生

简而言之 在官方推荐的插件中无法很好适配当前版本的nonebot 但是你们的文档仍旧把那些停止维护的插件放在推荐列表当中

需要经过用户安装后才会发现当前版本并不适配,或者无法安装

**Comment by coralZone** [comment]
需要经过用户安装后才会发现当前版本并不适配,或者无法安装

**Comment by HuanLinOTO** [comment]
https://nonebot.dev/store/plugins

群友说没有推荐列表 看了一下确实没有 是不是你看错了

**Comment by XTxiaoting14332** [comment]
插件名称旁边的插头不就是干这个的吗（）

**Comment by j1g5awi** [comment]
請您直接說明您認爲停止維护的插件是哪些

**Comment by coralZone** [comment]
你们自己去看项目的README 不够显眼吗

**Comment by XTxiaoting14332** [comment]
插件列表中也有指向NoneBot Registry的链接，用户可以通过这个链接查看插件支持的适配器和依赖

**Comment by HuanLinOTO** [comment]
![QQ_1721033288858](https://github.com/user-attachments/assets/f856bab2-3a84-4dc7-8a77-98cd18940a68)


**状态**？

**Comment by coralZone** [comment]
既然你们都觉得这是理所应当知道的 那我问这个问题也没什么意义

**Comment by HuanLinOTO** [comment]
可是你全过程都没有说明白问题 也没有引用任何一篇文档 readme

**Comment by sena-nana** [comment]
问题官方也没有推荐插件啊（

**Comment by coralZone** [comment]
> 可是你全过程都没有说明白问题 也没有引用任何一篇文档 readme

没有说过吗？

**Comment by RF-Tar-Railt** [comment]
请您放几张截图来说明您的问题

**Comment by HuanLinOTO** [comment]
> 你们自己去看项目的README 不够显眼吗



> ![QQ_1721033288858](https://private-user-images.githubusercontent.com/94176676/348646668-f856bab2-3a84-4dc7-8a77-98cd18940a68.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwMzM2ODcsIm5iZiI6MTcyMTAzMzM4NywicGF0aCI6Ii85NDE3NjY3Ni8zNDg2NDY2NjgtZjg1NmJhYjItM2E4NC00ZGM3LThhNzctOThjZDE4OTQwYTY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE1VDA4NDk0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmMmEyOGIxOGU5MzEwMWJjMjhjNTkyZTZlZDVjZWNiYmFlZGNkY2U3ZjE4M2I1MzBkNWY4NGJmYTk2NGU2Y2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Mb87kJB_zzXcB-v_rxBWlOhafsc_ek1V0LlQQfI7txk)
> 
> **状态**？

README 这是适配器 不是插件 如果你说的是别的仓库的 README 你得引用过来 没人想算卦猜谜

**Comment by coralZone** [comment]
![image](https://github.com/user-attachments/assets/c0f267d5-c578-4af0-b0e3-e0ee7cd145bc)
截图当中的内容是项目里面的README吧？
请问状态有具体的注解 或者释义吗？ 这些 icon 什么意思README里面有过说明吗

**Comment by XTxiaoting14332** [comment]
> ![image](https://github.com/user-attachments/assets/c0f267d5-c578-4af0-b0e3-e0ee7cd145bc)
> 截图当中的内容是项目里面的README吧？
> 请问状态有具体的注解 或者释义吗？ 这些 icon 什么意思README里面有过说明吗

？我寻思着右边的注释不就是

**Comment by sena-nana** [comment]
> ![image](https://private-user-images.githubusercontent.com/171648160/348647885-c0f267d5-c578-4af0-b0e3-e0ee7cd145bc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwMzM5NDAsIm5iZiI6MTcyMTAzMzY0MCwicGF0aCI6Ii8xNzE2NDgxNjAvMzQ4NjQ3ODg1LWMwZjI2N2Q1LWM1NzgtNGFmMC1iMGUzLWUwZWU3Y2QxNDViYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxNVQwODU0MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNDUwZTE4MjI5MjZiNmZiMmEyNDZkZGQ2N2FlNjA5Y2M5ZTM1YmM4ZWM5ODM5NjdiYjhlYTU0NTYyN2Q5NjczJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.yOmk_3k-UCSMl4X-ZvTvJ4Am4Vxq8iOlWKJfRZWLclE) 截图当中的内容是项目里面的README吧？ 请问状态有具体的注解 或者释义吗？ 这些 icon 什么意思README里面有过说明吗

首先这些不是插件

**Comment by KomoriDev** [comment]
这和你 title 中说的『插件』有什么关系

**Comment by HuanLinOTO** [comment]
如果真的要写 CI 测试 adapter 的话，没法写

只能测试可否成功 `import` 这点我没记错的话已经在上架 store 流程实现了

除非作者自己写了测试来判断过不过

转自群友

**Comment by HuanLinOTO** [comment]
另外你用的啥适配器

**Comment by coralZone** [comment]
一个微信的适配器

**Comment by HuanLinOTO** [comment]
NtChat 我看说已不再维护

这也不能赖官方吧，官方已经尽量避免上架的插件不能用的情况了

**Comment by AzideCupric** [comment]
在初次提出issue的时候就应该给出具体的无法使用的包的名称，这样能显著减少理解偏差

**Comment by RF-Tar-Railt** [comment, marked as answer]
> ![image](https://github.com/user-attachments/assets/c0f267d5-c578-4af0-b0e3-e0ee7cd145bc)
> 截图当中的内容是项目里面的README吧？
> 请问状态有具体的注解 或者释义吗？ 这些 icon 什么意思README里面有过说明吗

我这么跟你说，打绿√的都是官方维护性适配器，即使不更新也能保证可用性

标注了社区贡献的适配器是由贡献者自己pr在readme上面的，其可用性应当由贡献者自行更新和维护，官方没有义务去监管。

如果你觉得某个适配器应当下架或者修改状态，请你与适配器贡献者联系后自己开pr来更新

**Comment by coralZone** [comment]
或者在申请进入插件商店的时候需要附带testing case,每一个插件用例都添加单元测试

**Comment by HuanLinOTO** [comment]
pr welcome

**Comment by Lipraty** [comment]
PR‘s Welcom

**Comment by j1g5awi** [comment]
鑑於本 Issue 的回覆太過頻繁，轉換成 Discussion。

最後提醒各位，**Issue 不是即時聯天軟件**。在提出或回覆 Issue 時，你可以編輯之前的內容。

**Comment by coralZone** [comment]
你们作为熟于nonebot的开发者 当用户提出问题 或者是一些功能请求 都是希望使用者变成开发者吗？ 虽然开源社区确实是需要每一个人参与,但是参与是有门槛的。不是每一个人都有大量的时间去参与开源项目的。我提出issue就是希望项目维护者能够关注提问者的建议或问题。
但是把所有的问题都丢给新用户自己解决 那么issue的意义在哪里呢？

**Reply by coralZone** [reply]
或者说, 维护者希望只有维护者才来使用这个项目 否则 是没有资格使用的？

**Reply by RF-Tar-Railt** [reply]
由社区贡献的适配器，其可用性本就应当由开发者自觉更新。

贡献者自己只在readme提及适配器情况，未在nonebot2仓库中的readme列表内更新，这个问题适配器是贡献者自己造成的。

**Reply by j1g5awi** [reply]
开发群的人確實有些不够礼貌，但當我讓閣下直接指出問題所在，您回复的也称不上恰当，直到关闭 Issue 之后我才知道您所指的是 README 页面的社区适配器。

您如果还有微信适配器的问题，请移步该适配器的仓库。

您提到的单元测试问题，我们已经对社区插件进行[此类测试](https://registry.nonebot.dev/)，没有对社区适配器进行同种测试是我们的不足。
但说到底，本项目确实是一个开源项目，对于我们暂时无暇顾及的部分确实只能期望社区贡献。

**Comment by yanyongyu** [comment]
关于你提出的问题：

1. 官方仅对官方支持的适配器负责并对维护中的适配器保证可用性
2. 官方没有这个权限与能力去要求社区开发者开发内容的质量和可用性
3. 个人认为目前生态系统已较为完善，已经具有自动化测试且在商店中进行了标识
4. 如果你认为某个适配器或者插件导致了你无法使用，请你去联系对应作者而不是 NB 官方
